File: code/cpp/ch03/practice/challenge_01_diamond.cpp
/*
 * Challenge 01: Diamond Pattern
 * ==============================
 * Chapter 3: Decisions and Loops
 *
 * PROBLEM
 * -------
 * Given a positive integer n, return a string representing a diamond
 * pattern made of stars. The diamond has n rows in the top half
 * (including the middle/widest row) and n-1 rows in the bottom half.
 * Total rows = 2*n - 1. The widest row has 2*n - 1 stars.
 *
 * For n = 1:
 *   *
 *
 * For n = 2:
 *    *
 *   ***
 *    *
 *
 * For n = 3:
 *     *
 *    ***
 *   *****
 *    ***
 *     *
 *
 * Lines are separated by newlines. No trailing newline.
 * No trailing spaces on any line.
 *
 * INPUT FORMAT
 * ------------
 * A single line containing a positive integer n.
 *
 * OUTPUT FORMAT
 * -------------
 * Print the diamond pattern.
 *
 * CONSTRAINTS
 * -----------
 * 1 <= n <= 25
 *
 * EXAMPLES
 * --------
 * Input:  1
 * Output: *
 *
 * Input:  2
 * Output:
 *  *
 * ***
 *  *
 *
 * INSTRUCTIONS
 * ------------
 * Replace the body of solve() with your solution.
 * The main() function handles I/O -- don't change it.
 */

#include <iostream>
#include <string>
using namespace std;

/**
 * Return a diamond pattern string with n as the half-height (including center).
 * Rows separated by '\n'. No trailing newline or trailing spaces.
 */
string solve(int n) {
    // TODO: Replace this with your solution
    return "";
}

// ── Do not change anything below this line ──────────────────────────
int main() {
    int n;
    cin >> n;
    cout << solve(n) << endl;
    return 0;
}

File: code/python/ch03/practice/challenge_01_diamond.py
"""
Challenge 01: Diamond Pattern
==============================
Chapter 3: Decisions and Loops

PROBLEM
-------
Given a positive integer n, return a diamond pattern of stars. The top
half (including the middle row) has n rows. The total height is 2n - 1.
Each row has leading spaces followed by an odd number of stars.
No trailing spaces on any row. Rows separated by newlines.
No trailing newline.

INPUT FORMAT
------------
A single line containing a positive integer n.

OUTPUT FORMAT
-------------
Print the diamond pattern.

CONSTRAINTS
-----------
1 <= n <= 50

EXAMPLES
--------
Input:  3
Output:
  *
 ***
*****
 ***
  *

Input:  1
Output:
*

Input:  2
Output:
 *
***
 *

INSTRUCTIONS
------------
Replace the `pass` in the solve() function with your solution.
The main block at the bottom handles input/output — don't change it.
"""


def solve(n: int) -> str:
    """Return a diamond pattern of stars with n rows in the top half."""
    pass  # TODO: Replace this with your solution


# ── Do not change anything below this line ──────────────────────────
if __name__ == "__main__":
    n = int(input())
    print(solve(n))
